charts/elasticsearch/templates/backup/cronjob.yaml

```diff
@@ -0,0 +1,36 @@
+{{- if .Values.backup.enabled }}
+apiVersion: batch/v1beta1
+kind: CronJob
+metadata:
+  name: {{ .Values.name }}-backup-cronjob
+  labels: {{ include "elasticsearch.labels" . | nindent 4 }}
+spec:
+  schedule: "0 8 * * *" # Backup every day at 3 am EST
+  concurrencyPolicy: Replace
+  jobTemplate:
+    metadata:
+      labels: {{ include "elasticsearch.labels" . | nindent 8 }}
+    spec:
+      activeDeadlineSeconds: {{ .Values.backup.timeoutSeconds }}
+      template:
+        metadata:
+          labels: {{ include "elasticsearch.labels" . | nindent 12 }}
+        spec:
+          serviceAccountName: {{ .Values.name }}-backup-sa
+          restartPolicy: OnFailure
+          nodeSelector: {{ .Values.backup.nodeSelector | toYaml | nindent 12 }}
+          tolerations: {{ .Values.backup.tolerations | toYaml | nindent 12 }}
+          containers:
+          - name: {{ .Values.name }}-backup
+            image: {{ .Values.backup.imageRepo}}:{{ .Values.backup.imageTag }}
+            command: ['sh', '-c']
+            args:
+            - |-
+              TIMESTAMP=$( date "+%Y%m%d.%H%M%S" ) && \
+              curl -X PUT "${ELASTICSEARCH_HOSTNAME}:9200/_snapshot/${SNAPSHOT_REPOSITORY}/es-snapshot-${TIMESTAMP}?wait_for_completion=true"
```

Comment:
If you append `:?` to your Bash variable name references, you'll get an error with a default message if the variable is null or unset.

Reply:
TIL again 👍🏻